### PR TITLE
FieldOverides: use the registry to process field values

### PR DESCRIPTION
**Pull request by ryantxu**
Use the registry `process` method to set values

### CODE REVIEW

File: packages/grafana-ui/src/components/FieldConfigs/fieldOverrides.test.ts [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+      theme: (undefined as any) as GrafanaTheme,
+    })[0];
+    const valueColumn = data.fields[1];
+    const config = valueColumn.config;
+
+    // Keep max from the original setting
+    expect(config.max).toEqual(0);
+
+    // Don't Automatically pick the min value
+    expect(config.min).toEqual(-20);
```
Comment (dprokop) [inline packages/grafana-ui/src/components/FieldConfigs/fieldOverrides.test.ts]: @ryantxu get this test failing and failing over again. Not sure this assertion is actually right. Should the min value be derived from data here?

Comment (dprokop) [reply packages/grafana-ui/src/components/FieldConfigs/fieldOverrides.test.ts]: Solved!

File: packages/grafana-data/src/types/dataFrame.ts [hunk 2]
```diff
@@ -50,6 +51,8 @@ export interface FieldConfig {
 
   // Panel Specific Values
   custom?: Record<string, any>;
+
+  scopedVars?: ScopedVars;
```
Comment (torkelo) [inline packages/grafana-data/src/types/dataFrame.ts]: why is scopedVars here?

**Review by torkelo — APPROVED** [review]
Looks really good. Will have to walk me through this next week :) 

only thing that was a bit confusing was the scoped vars on field config

File: packages/grafana-data/src/types/dataFrame.ts (hunk 2 above)
Comment (dprokop) [reply packages/grafana-data/src/types/dataFrame.ts]: Added those for the overrides processor to be able to interpolate field variables. See https://github.com/grafana/grafana/pull/22047/files#diff-3b41a4dd550fb20fd66edb288940b5c9R17 for example. So particular field override config can specify whether or not variables should be interpolated, hence we need to have access to the vars

**Pull request merged**